cfile/CMakeLists.txt

```diff
@@ -1,7 +1,6 @@
-set(HEADERS )
 set(CPPS
-		CFILE.cpp
-		hog.cpp
-		InfFile.cpp)
+	cfile.cpp
+	hogfile.cpp
+	inffile.cpp)
 
-add_library(cfile STATIC ${HEADERS} ${CPPS})
\ No newline at end of file
+add_library(cfile STATIC ${CPPS})
```

Comment:
To avoid adding the root of the project to find the "cfile/cfile.h" include, you can do:

1. Revert all the includes to be `#include <cfile.h>` instead of `#include "cfile/cfile.h"`
2. Here have `target_include_directories(cfile PUBLIC .)`. That will make any dependent of the `cfile` library find its includes.

Reply:
Main idea about it is splitting code into manageable submodules that can be accessed from root of project. Maybe ${CMAKE_SOURCE_DIR} is not good for it, but ${PROJECT_SOURCE_DIR} would better?

Reply:
No, I'd still depend on the library to provide the includes instead of a blank `include_directories` (which I'd remove completely as well in favor of `target_include_directories` for each, but that's a separate discussion), I think that's the correct semantic for that (you depend on the assets of the `cfile` library, so it makes sense you add a dependency on it)

Reply:
I've added BUILD_INTERFACE to cfile module. Other modules that depends on it, can reuse includes on linking.
There some files formally not belonging any packages (lib directory), as workaround there temporary include_directories(cfile) on root of project. After migrating all modules this can be removed.